src/runAll.js

```diff
@@ -54,15 +54,21 @@ module.exports = function runAll(config) {
           return `No staged files match ${task.pattern}`
         }
         return false
-      }
+      },
+      hasStagedFiles: () => task.fileList.length > 0
     }))
 
     const listrBaseOptions = {
       dateFormat: false,
       renderer
     }
 
-    if (tasks.length) {
+    let hasStagedTaskFiles = false
+    tasks.forEach(task => {
+      hasStagedTaskFiles = hasStagedTaskFiles || task.hasStagedFiles()
```

Comment:
I’m not sure I understand how this is working. Mind explaining in code comments?

Reply:
Changed the logic of this slightly in my next commit but also added a comment as you requested.